src/core/model/types/builtin.h

```diff
@@ -47,12 +48,14 @@ BETTER_ENUM(TypeId, char,
                  * (arbitrary-precision integer value) */
     kString,    /* Except for nulls and empties column contains only strings
                  * (string value, sequence of characters) */
+    kDate,      /* Except for nulls (not_a_date_time) and infinities(pos_infin, neg_infin) column contains only dates
```

Comment:
This enum describes both types of values which can be stored in the table and types of the table columns.
This PR adds date as a type for a single value, but lacks support for `Date` table columns. The code which creates `TypedColumnData` knows nothing about newly added `Date` and this is to be fixed. Please, look at the `TypedColumnDataFactory::CreateFrom` and implement the corresponding functionality. I suppose adding the appropriate values to `TypedColumnDataFactory::type_id_to_regex_` should be enough

Reply:
It seems to me that checking whether string value represents a date via regular expression is not a good option because there are some instances that will satisfy regular expression but are not a valid date (e.g. «2023-02-29») in such cases `Date` is constructed as `not_a_date_time` so i think it would be better if we just leave it as a string value. `boost::gregorian` doesn’t provide any option to check whether string value represents a valid date without constructing a `Date` object so i don’t think this is a good option either. Is there any way to check it via regular expression more precisely (i use `^([0-9]{4})[-./](1[0-2]|0[1-9])[-./](3[0-1]|0[1-9]|[1-2][0-9])$` expression from "Regular Expressions Cookbook»)  or should i just leave it as it is?